bin.js

```diff
@@ -0,0 +1,30 @@
+#! /usr/bin/env node
+
+const { promisify } = require("util");
+const { exec } = require("child_process");
+const execAsync = promisify(exec);
+
+const execute = async (command, showSuccess) => {
+    // Generate access token
+    const { stderr, stdout } = await execAsync(command);
+    if (stderr) {
+        throw stderr;
+    }
+
+    if (showSuccess && stdout) {
+        console.log(stdout);
+    }
+}
+
+const run = async () => {
+    console.log("Retrieving your Cloudflare access token. Please head to your browser to approve");
+    await execute("cloudflared access login https://bitbucket.ship.gov.sg", true);
+
+    console.log("Applying access token to Bitbucket headers");
+    await execute(`git config --global --replace-all http.https://bitbucket.ship.gov.sg/.extraheader "cf-access-token: $(cloudflared access token -app=https://bitbucket.ship.gov.sg)"`);
+
+    await execute("export cloudflared_access_token=$(cloudflared access token -app=https://bitbucket.ship.gov.sg)");
```

Comment:
this wouldn't work because the env is set in a child process, so the current shell still doesn't have access to $cloudflared_access_token

but haven't found any good solutions for this online

have seen a suggestion for `eval ${<command>)` instead but that requires the command to have very specific output to stdout

Reply:
Is there a particular reason why the script needs to be translated to a js file?
We shouldn't run into this, if it's just the shell script, right?

Reply:
even with just the shell script, you'll need to call it with `source` or `.`

Reply:
On discussion with @qroll we'll just do the login process in this script to allow users to continue to `npm ci`. But they will need to run another script from mol-lib-config to apply the token to the env for the docker-compose

Reply:
Hmm were you trying to run the .sh with npm and it didn't work, or run directly without npm and it didn't work?